helper.php

```diff
@@ -364,6 +386,156 @@ function _formatData($column, $value, Doku_Renderer_xhtml $R) {
     }
 
     /**
+     * Return formated data, depending on column type
+     *
+     * @param array               $column
+     * @param string              $value
+     * @param Doku_Renderer       $R
+     * @return string
+     */
+    function _formatDataNew($column, $value, Doku_Renderer $R) {
+        global $conf;
+        $vals = explode("\n", $value);
+        $outs = array();
+
+        //multivalued line from db result for pageid and wiki has only in first value the ID
+        $storedID = '';
+
+        foreach($vals as $val) {
+            $val = trim($val);
+            if($val == '') continue;
+
+            $type = $column['type'];
+            if(is_array($type)) {
+                $type = $type['type'];
+            }
+            switch($type) {
+                case 'page':
+                    $val = $this->_addPrePostFixes($column['type'], $val);
+                    $val = $this->ensureAbsoluteId($val);
+                    $outs[] = $R->internallink($val, null, null, true);
+                    break;
+                case 'title':
+                    list($id, $title) = explode('|', $val, 2);
+                    $id = $this->_addPrePostFixes($column['type'], $id);
+                    $id = $this->ensureAbsoluteId($id);
+                    $outs[] = $R->internallink($id, $title, null, true);
+                    break;
+                case 'pageid':
+                    list($id, $title) = explode('|', $val, 2);
+
+                    //use ID from first value of the multivalued line
+                    if($title == null) {
+                        $title = $id;
+                        if(!empty($storedID)) {
+                            $id = $storedID;
+                        }
+                    } else {
+                        $storedID = $id;
+                    }
+
+                    $id = $this->_addPrePostFixes($column['type'], $id);
+
+                    $outs[] = $R->internallink($id, $title, null, true);
+                    break;
+                case 'nspage':
+                    // no prefix/postfix here
+                    $val = ':' . $column['key'] . ":$val";
+
+                    $outs[] = $R->internallink($val, null, null, true);
+                    break;
+                case 'mail':
+                    list($id, $title) = explode(' ', $val, 2);
+                    $id = $this->_addPrePostFixes($column['type'], $id);
+                    if(!$title) {
+                        $title = $id;
+                    } else {
+                        $title = hsc($title);
+                    }
+                    $outs[] = $R->emaillink($id, $title, true);
```

Comment:
Ich kehre zur�ck am 10.08.2015.

Ich werde Ihre Nachricht nach meiner R�ckkehr beantworten. Herzlichen Dank
f�r ihr Verst�ndnis.

I will answer your message after returning.


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[dokuwiki-plugin-data] Support for odt export in datatable and datalist
(#169)" gesendet am 30.07.2015 22:50:48.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, w�hrend
diese Person abwesend ist.